### PR TITLE
Gealber PCC49, need some changes

### PR DESCRIPTION
**Pull request by Gealber**
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): [8]
Estimated time spent (hours): [3]
Completed (yes/no): [yes]
I stretched my coding skills (if yes what did you learn?): [Learned to use the queue module of Python Standard Library and a little bit more about web scrapping. This is fascinating]
Other feedback (what can we improve?): []

### CODE REVIEW

File: 49/Gealber/crawl_verifier.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        If there's an error when parsing the url is added to error_links set.
+        Otherwise extract the links of the page with BeautifulSoup and pass this links to the method
+        enqueue_links() to well ....be enqueued.
+        :param thread_name: Current thread name
+        :param url: url to connect
+        :param links_to_crawl: queue with the links extracted.
+        """
+        try:
+            if urlparse(url).netloc == 'planetpython.org' and url not in Crawl.error_links:
+                req = Request(url, headers=Crawl.headers)
```
Comment (bbelderbos) [inline 49/Gealber/crawl_verifier.py:42]: any reason to not go with `requests`?

File: 49/Gealber/crawl_verifier.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        """
+        This method connect to the given url and put it in the corresponding set() or queue.
+        If there's an error when parsing the url is added to error_links set.
+        Otherwise extract the links of the page with BeautifulSoup and pass this links to the method
+        enqueue_links() to well ....be enqueued.
+        :param thread_name: Current thread name
+        :param url: url to connect
+        :param links_to_crawl: queue with the links extracted.
+        """
+        try:
```
Comment (bbelderbos) [inline 49/Gealber/crawl_verifier.py:40]: make this try block narrower, `requests` would provide you with a `raise_for_status` = convenient

File: 49/Gealber/crawl_verifier.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        with urlopen(req, timeout=timeout) as resp:
+            page = str(resp.read(), 'utf-8')
+
+        soup = BeautifulSoup(page, 'lxml')
+        children = soup.select("ul.level-one > li")
+        extract_sections = []
+        for child in children:
+            extract_sections.append(child)
+        links = []
+        for i in range(1, 4):
```
Comment (bbelderbos) [inline 49/Gealber/crawl_verifier.py:113]: put magic numbers in constants

File: 49/Gealber/crawl_verifier.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    Well...........
+    :param url:
+    """
+    try:
+        Crawl.crawl(threading.current_thread(), url, links_to_crawl)
+        logging.info(f"Starting to crawl {url} in thread {threading.current_thread()}")
+        print("Wasiting 5 secs...")
+        logging.info("Waiting 5 secs....")
+        time.sleep(5)
+    except:
```
Comment (bbelderbos) [inline 49/Gealber/crawl_verifier.py:136]: try to avoid bare exceptions

File: 49/Gealber/crawl_verifier.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    }
+    url = 'http://planetpython.org'
+    Crawl(headers=headers, base_url=url)
+    links_to_crawl.put(url)
+    inspection = Inspection(headers=headers)
+    urls_to_inspection = [url for url in inspection.read_config('config.ini')]
+    urls_to_inspection.append(inspection.links_start_page(timeout=60))
+    for link in urls_to_inspection:
+        links_to_crawl.put(link)
+    while not links_to_crawl.empty():
```
Comment (bbelderbos) [inline 49/Gealber/crawl_verifier.py:160]: break down into multiple functions

File: 49/Gealber/crawl_verifier.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+from urllib.parse import urljoin, urlparse
+from bs4 import BeautifulSoup
+import sqlite3
+import feedparser
+from datetime import date
+import queue
+import time
+import logging
+
+conn = sqlite3.connect('urls.db')
```
Comment (bbelderbos) [inline 49/Gealber/crawl_verifier.py:13]: global scope, put in class?

File: 49/Gealber/crawl_verifier.py (hunk 1 above)
Comment (Gealber) [reply 49/Gealber/crawl_verifier.py:42]: When I wrote the code I only new urllib. But yes with requests would much more simpler and cleaner

**Pull request merged**